### ISSUE TITLE
Long-press button to (temporarily) black out screen

### ISSUE THREAD

**Issue by youri--** [theothernt/AerialViews#101]
tl;dr: see issue title

I have my TV on a lot, often playing music, with Aerial Views for the visuals. There are times however, I would like a black screen, nothing displayed, without actually turning off the TV.

For example:

* I'm busy doing stuff, can't watch the screen, I only need music;
* I sometimes leave soft music playing to sooth my dogs, while I'm gone.

In these occasions, with modern LED/OLED TVs, displaying a black screen would save energy. And saving energy is good for my wallet and the environment.

SmartTubeNext recently added a setting that blacks out the screen after a fixed amount of time. I (ab)use this feature in the aforementioned situations, by opening STN and letting it sit there, but how cool would it be, if I could manually trigger a black screen from my actual screensaver (which I can already also trigger to start, with a double-tap on my Nvidia Shield's Settings button).

I was thinking (optional, a new setting) a long-press on the OK/Select button (the dpad-center button) would be great. Then another tap on any dpad button to cancel the blacked out screen and return to the beautiful video visuals.

**Comment by theothernt** [comment]
That sounds like a good idea! 

As you suggest, it will be an opt-in feature on the `Settings > D-Pad/Remote` screen. When Aerial Views is running, you can do the following...

1. Long-press the OK button the fade out and stop the video
2. Press OK to re-start video

Do you think the other directional buttons should be disabled when then screen is off?

**Comment by Mr-HaleYa** [comment]
> Yes, that's probably a good idea.

Could cause frustration when someone other than you tries to use the remote when in this mode. The most likely outcome is going to be them turning the TV off and then back on when none of the buttons work.

Also would be nice to have a toggle in this mode settings to add the clock in like with the videos.

**Comment by theothernt** [comment]
> Could cause frustration when someone other than you tries to use the remote when in this mode. The most likely outcome is going to be them turning the TV off and then back on when none of the buttons work.

Understood, there will be a couple of options to configure - ie. hold OK to sleep, press OK to wait or any button/direction

> Also would be nice to have a toggle in this mode settings to add the clock in like with the videos.

Do you mean to show and hide overlays (ie. clock, location, date, etc)?

**Comment by theothernt** [comment]
> For me, just a blank screen would be enough, but I can see how, as an additional option toggle, other overlay options like the clock could be nice to have, so long as the overlays are still moved around (to prevent oled burn-in) every few minutes.

That's the plan for this feature. v1.7.2 will have the ability to customise all directions + press or long press. If that works well, I'll add the 'blank screen' option for v1.7.3.

> IIRC the moving around now happens when videos are switched? If so, then on a blank screen it would need to be on a timer instead, say 3.minutes?

Currently, the overlays switch/alternate sides after every video or photo change (anywhere from 30 seconds to 10+ mins). If the screen is blank, the overlays will be hidden as well.

> Thank you for not forgetting old feature requests, @theothernt 🙏

I try to get there eventually with most features 😅🤞🏻

**Comment by theothernt** [comment]
Here is [v1.7.3 beta1 for testing](https://www.dropbox.com/scl/fi/4jjbxnojc8w8y2b02ijz7/aerial-views-1.7.3-beta1.apk?rlkey=f6efwndvt5rtb7ynmwlv776ay&dl=0). It includes...

* Black out mode which hides the media and overlays
* Change volume of videos

Let me know if a) anything is unclear or b) you find bugs or issues.

**Comment by theothernt** [comment]
> I have played around with blackout screen functionality for a bit and so far it works great. I've tried different buttons, both short and long-press and cancelling with any button or the button(s) assigned to screen blackout (I also assigned it to multiple for testing, starting blackout with one key and ending it with another). All good. 👌

Glad to hear different button combos are working, that was a worry initially but it seems to be working for everyone.

> I have 1 note: I'm going to assign blackout screen to long-press OK button (center dpad on my shield tv), and I noticed that with "Wake from blackout mode with any button" turned off, it will then also only turn off with a long-press. I would suggest to also allow wake with a short press of the assigned button.

Yes, so you have two options - wake with any button/direction press OR wake with the button/direction that started black out mode. So you're suggesting a 3rd wake option?

> This is also the first time I notice the "music playing" overlay displaying the song info from the Radio Paradise app, awesome!

Glad to hear that's working, there was a bug in 1.7.2 which meant Now Playing doesn't work for some devices, but it's been fixed now.

> 1. The beta wouldn't install as update, so I had to uninstall it first and hence reconfigure everything. I think the app at this point has so many things you can configure, it could really use a settings export/import. Just like for example a JSON or xml file to/from local storage (Download folder, or even internal storage/data/<app ID or package name> like what SmartTube uses).

Sorry about that. The Play Store and Github version, although they have the same app name, user a different security cert, so they are 2 different apps to Android. Also, I don't have a setting import/export feature yet as Android 10/11/12+ have changed how apps access files... so it's something I need to figure out!

> 2. Going into any settings menu and then a submenu, then Back, you always end up back at the top. It would be a huge quality of life feature, when hitting Back, for the selection to be at the submenu you were just at. E.g. I enter the "D-Pad/Remote" submenu, hit Back and the top item "Overlays" is selected instead of the "D-Pad/Remote" submenu I was just in.

That's a long running issue I'd like to fix, but I'm not sure how. The settings screen, from Android, is quite old and made for phones so I think the only option is to re-make the entire UI but I'm not sure when I'll be able to do that just yet.

**Comment by theothernt** [comment]
v1.7.3 has just been released on GitHub with this feature, it should appear on the Play Store in a few days assuming no issues.

**Comment by theothernt** [comment]
> 2. Going into any settings menu and then a submenu, then Back, you always end up back at the top. It would be a huge quality of life feature, when hitting Back, for the selection to be at the submenu you were just at. E.g. I enter the "D-Pad/Remote" submenu, hit Back and the top item "Overlays" is selected instead of the "D-Pad/Remote" submenu I was just in.

So... you'll never guess... I actually fixed this issue. It took a week of development with many dead ends, but it's fixed!

I do have to test it, make sure it's stable... but if it is, it'll be in the next release! 😅🎉

**Comment by theothernt** [comment]
> My suggesting is that when you've assigned black out to a long-press button, and the option to wake with any button is off, to then simply allow the wake with the chosen button with any press (short or long). So not really a 3rd option. That way when I've activated black-out with long-press OK and someone else needs to turn the screen on (and doesn't remember this feature or that it needs to be a long-press), they can just press OK (still only that button, but short/long doesn't matter). 
Sorry, hope that makes sense :)

I think that's the way the feature works now, let me know!
 
> No problem, no export/import option is not a huge deal, though it would be a (very) nice-to-have.

I'm still trying to find a solution to this. The basic problem is, if AV creates a text file, you uninstall the app, you install the app again at a later date, AV cannot read that text file. It's very stupid.

And this doesn't happen on the phone as it has a built-in file picker, which the TV does not. Again, very stupid.

**Comment by theothernt** [comment]
> My suggesting is that when you've assigned black out to a long-press button

This has been live for a couple of versions now, is the feature working ok for you?

> No problem, no export/import option is not a huge deal, though it would be a

I've found a method that will work. It should be in the next release, v1.7.5.

Backing up the settings will be straight forward. Restoring them will require a File Manager app to 'share' the settings file with Aerial Views.
